src/kokkos_abstraction.hpp

```diff
@@ -976,6 +976,65 @@ KOKKOS_FORCEINLINE_FUNCTION void par_for_inner(team_mbr_t team_member, Args &&..
   par_for_inner(DEFAULT_INNER_LOOP_PATTERN, team_member, std::forward<Args>(args)...);
 }
 
+// Inner reduction loops
+template <typename Function, typename T>
+KOKKOS_FORCEINLINE_FUNCTION void
+par_reduce_inner(team_mbr_t team_member, const int kl, const int ku, const int jl,
+                 const int ju, const int il, const int iu, const Function &function,
+                 T reduction) {
```

Comment:
Should these be 
```c++
par_reduce_inner(InnerLoopPatternTTR, team_mbr_t team_member, const int kl, const int ku, const int jl,
                 const int ju, const int il, const int iu, const Function &function,
                 T reduction) {
```

Reply:
I'm not sure. I didn't really think about it when dropping these in, and I don't think I fully understand the way loop pattern resolution is done anyway...
The way I understand it, since I use `TVR_INNER_LOOP` for `ThreadVectorRange`, these wouldn't resolve under that, right?  So, we'd need a TVR and SIMD pattern for these too, if we wanted to make them selectable.  Since these should be used pretty rarely, I don't know if the performance benefit of allowing customization is worth three implementations we're just going to stamp over with all the better general stuff you guys are writing for the next release.

Reply:
I could see it both ways. With my suggestion, you would have to call `parthenon::par_reduce_inner(inner_loop_pattern_ttr_tag, ...)` instead of `parthenon::par_reduce_inner(...)`, which would be more explicit that you aren't necessarily using the default inner loop pattern and conform to how the rest of the `par_*_inner` functions look. But this doesn't change anything about how this behaves.

Reply:
Ah, right, you can explicitly tag loops! It seems better to be explicit, then, let me give it a go today.